File: app/src/main/java/com/melabear/newcomposedemo/CoffeeLogger.java
package com.melabear.newcomposedemo;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class CoffeeLogger {
    private final List<String> logs = new ArrayList<>();

    @Inject
    public CoffeeLogger() {
        CoffeeApp.print("Creating CoffeeLogger: " + this/*, true*/);
    }

    public void log(String msg) {
        logs.add(msg);
    }

    public List<String> logs() {
        return new ArrayList<>(logs);
    }
}

File: app/src/main/java/com/melabear/newcomposedemo/CoffeeLogger.kt
package com.melabear.newcomposedemo

import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class CoffeeLogger @Inject constructor() {
    private val logs: MutableList<String> = ArrayList()

    init {
        CoffeeApp.print("Creating CoffeeLogger: $this" /*, true*/)
    }

    fun log(msg: String) {
        logs.add(msg)
    }

    fun logs(): List<String> {
        return ArrayList(logs)
    }
}